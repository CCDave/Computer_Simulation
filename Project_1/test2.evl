module top;
	
	wire s1, s2, out, out2;
	adder out, s1, s2; //this is an adder
	adder out2, s1, out;
	
endmodule