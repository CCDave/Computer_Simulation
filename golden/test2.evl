// Testing multi-bit wires
module top;	
	wire [4:0] a2;
	output dff(a1,a2);
	dff(a1[0],a2[4]);
	wire [4:0] a1;
	dff(a1[2:0],a2[4:2]);
endmodule