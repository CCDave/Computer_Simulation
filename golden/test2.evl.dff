2
pin 5 a1[0] a1[1] a1[2] a1[3] a1[4]
pin 5 a2[0] a2[1] a2[2] a2[3] a2[4]
