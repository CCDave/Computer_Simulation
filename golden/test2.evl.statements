STATEMENT 1 MODULE, 3 tokens
STATEMENT 2 WIRE, 8 tokens
STATEMENT 3 COMPONENT, 8 tokens
STATEMENT 4 COMPONENT, 13 tokens
STATEMENT 5 WIRE, 8 tokens
STATEMENT 6 COMPONENT, 17 tokens
STATEMENT 7 ENDMODULE
