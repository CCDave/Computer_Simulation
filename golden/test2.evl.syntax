module top 2 3
wire a1 5
wire a2 5
component output dff 2
pin a1 4 0
pin a2 4 0
component dff NONE 2
pin a1 0 0
pin a2 4 4
component dff NONE 2
pin a1 2 0
pin a2 4 2
