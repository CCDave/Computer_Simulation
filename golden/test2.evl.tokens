NAME module
NAME top
SINGLE ;
NAME wire
SINGLE [
NUMBER 4
SINGLE :
NUMBER 0
SINGLE ]
NAME a2
SINGLE ;
NAME output
NAME dff
SINGLE (
NAME a1
SINGLE ,
NAME a2
SINGLE )
SINGLE ;
NAME dff
SINGLE (
NAME a1
SINGLE [
NUMBER 0
SINGLE ]
SINGLE ,
NAME a2
SINGLE [
NUMBER 4
SINGLE ]
SINGLE )
SINGLE ;
NAME wire
SINGLE [
NUMBER 4
SINGLE :
NUMBER 0
SINGLE ]
NAME a1
SINGLE ;
NAME dff
SINGLE (
NAME a1
SINGLE [
NUMBER 2
SINGLE :
NUMBER 0
SINGLE ]
SINGLE ,
NAME a2
SINGLE [
NUMBER 4
SINGLE :
NUMBER 2
SINGLE ]
SINGLE )
SINGLE ;
NAME endmodule
