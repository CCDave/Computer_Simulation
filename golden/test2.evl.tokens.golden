NAME module
NAME top
SINGLE ;
NAME wire
NAME s1
SINGLE ,
NAME s2
SINGLE ,
NAME out
SINGLE ,
NAME out2
SINGLE ;
NAME adder
NAME out
SINGLE ,
NAME s1
SINGLE ,
NAME s2
SINGLE ;
NAME adder
NAME out2
SINGLE ,
NAME s1
SINGLE ,
NAME out
SINGLE ;
NAME endmodule
